packages/contracts/contracts/optimistic-ethereum/OVM/predeploys/OVM_L1MessageSender.sol

```diff
@@ -20,24 +17,24 @@ import { iOVM_ExecutionManager } from "../../iOVM/execution/iOVM_ExecutionManage
  * Compiler used: solc
  * Runtime target: OVM
  */
-contract OVM_L1MessageSender is iOVM_L1MessageSender {
-
-    /********************
-     * Public Functions *
-     ********************/
-
-    /**
-     * @return _l1MessageSender L1 message sender address (msg.sender).
-     */
-    function getL1MessageSender()
-        override
-        public
-        view
-        returns (
-            address _l1MessageSender
-        )
-    {
-        // Note that on L2 msg.sender (ie. evmCALLER) will always be the Execution Manager
-        return iOVM_ExecutionManager(msg.sender).ovmL1TXORIGIN();
+contract OVM_L1MessageSender {
+    constructor() {
+        // This is a bit of a hack. By using the low-level assembly `return` we can dictate the
```

Comment:
I wouldn't describe this as a hack, its a good solution to use the new opcode without needing to have a compiler fork